### PR TITLE
Add iree_vm_value_{none,i8,i16} functions.

### PR DESCRIPTION
**Pull request by bjacob**
The _none function leaves the data union uninitialized. If you want something else, please specify what. Do we want to zero the full 8 bytes?

### CODE REVIEW

**Comment by benvanik** [comment]
Generally agreed for user-visible things - but in this case the storage is an implementation detail of the type. Fixing the https://github.com/google/iree/issues/5542 TODO will prevent users from misusing the internal storage is better than initialized to 0 as that also has heisenbug potential and is hard to differentiate valid usage from invalid usage ("is the zero I read out something intended or a default initializer? if initialized to i16 0xFF00 and read as i8 0x00, is that part of the real value or an default initialized one, etc). If you have a variant with a type and then don't check or incorrectly map the type then you have a bug in your code and you want to know about that via loud errors (asserts/etc).
Basically, if it's invalid behavior we should be asserting instead of relying on default initialization - the 0 initialization is a safety net for broken code, not something that should be considered intentional behavior.

As for !NDEBUG, ASAN should be able to catch the use of uninitialized stack memory otherwise. If the user-visible API asserts then the !NDEBUG init is most useful for people like me who use debuggers and don't want to see garbage when looking at structs.
